File: 07-04-network-image-viewmodel/java-07-04-network-image-viewmodel/app/src/main/java/com/example/viewModelImage/DetailActivity.java
package com.example.viewModelImage;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModelProvider;

import com.example.viewModelImage.databinding.ActivityDetailBinding;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class DetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityDetailBinding binding =
                DataBindingUtil.setContentView(this, R.layout.activity_detail);
        ViewModel viewModel = new ViewModelProvider(this).get(ViewModel.class);
        binding.setViewModel(viewModel);
        binding.setLifecycleOwner(this);

        Intent intent = getIntent();
        int pid = intent.getIntExtra("pid", 1);
        viewModel.getPokemon(pid);
    }

    public static class ViewModel extends androidx.lifecycle.ViewModel {
        public MutableLiveData<PokemonResponse> response = new MutableLiveData<>();

        public void getPokemon(int pid) {
            Repository.getInstance().getPokemon(pid).enqueue(new Callback<PokemonResponse>() {
                @Override
                public void onResponse(Call<PokemonResponse> call, Response<PokemonResponse> response) {
                    PokemonResponse pokemonResponse = response.body();
                    ViewModel.this.response.setValue(pokemonResponse);
                }

                @Override
                public void onFailure(Call<PokemonResponse> call, Throwable t) {
                }
            });
        }
    }
}

File: 07-04-network-image-viewmodel/kotlin-07-04-network-image-viewmodel/app/src/main/java/com/example/viewModelImage/DetailActivity.kt
package com.example.viewModelImage

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModelProvider
import com.example.pagingWithDatabinding.R
import com.example.pagingWithDatabinding.databinding.ActivityDetailBinding
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class DetailActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityDetailBinding>(this, R.layout.activity_detail)
        val viewModel = ViewModelProvider(this).get(ViewModel::class.java)
        binding.viewModel = viewModel
        binding.lifecycleOwner = this

        val intent = intent
        val pid = intent.getIntExtra("pid", 1)
        viewModel.getPokemon(pid)
    }

    class ViewModel : androidx.lifecycle.ViewModel() {
        var response = MutableLiveData<PokemonResponse>()

        fun getPokemon(pid: Int) {
            Repository.getPokemon(pid).enqueue(object : Callback<PokemonResponse?> {
                override fun onResponse(call: Call<PokemonResponse?>, response: Response<PokemonResponse?>) {
                    val pokemonResponse = response.body()
                    this@ViewModel.response.value = pokemonResponse
                }

                override fun onFailure(call: Call<PokemonResponse?>, t: Throwable) {}
            })
        }
    }
}